codes/go/pkg/tree_node.go

```diff
@@ -25,6 +25,24 @@ func NewTreeNode(v int) *TreeNode {
 }
 
 // ArrToTree Generate a binary tree given an array
+// 序列化编码规则请参考：
+// https://www.hello-algo.com/chapter_tree/array_representation_of_tree/
+// 二叉树的数组表示：
+// [1, 2, 3, 4, None, 6, 7, 8, 9, None, None, 12, None, None, 15]
+// 二叉树的链表表示：
+//
+//	         /——— 15
+//	     /——— 7
+//	 /——— 3
+//	|    \——— 6
+//	|        \——— 12
+//
+// ——— 1
+//
+//	\——— 2
+//	   |    /——— 9
+//	    \——— 4
+//	        \——— 8
```

Comment:
The ArrToTree and TreeToArr should be updated to the new implementation. Please refer to Java Python C++

Reply:
done